src/Common.cpp

```diff
@@ -0,0 +1,29 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#include "src/Common.h"
+#include <glog/logging.h>
+#include <folly/io/IOBuf.h>
+#include "src/AbstractStreamAutomaton.h"
+
+namespace reactivesocket {
+
+static const char* getTerminatingSignalErrorMessage(int terminatingSignal) {
+  switch ((StreamCompletionSignal)terminatingSignal) {
+    case StreamCompletionSignal::CONNECTION_END:
+      return "connection closed";
+    case StreamCompletionSignal::CONNECTION_ERROR:
+      return "connection error";
+    case StreamCompletionSignal::ERROR:
+      return "general error";
+    case StreamCompletionSignal::GRACEFUL:
+      DCHECK(false) << "throwing exception for GRACEFUL termination?";
```

Comment:
Can this still return a string saying "graceful termination," just in case we need to debug it in production?